tensorrt_llm/_torch/pyexecutor/perf_metrics_manager.py

```
@@ -281,3 +346,105 @@ def append_step_metrics(self, request, iter_counter: int, batch_token_time=None)
+            self._events_file.close()
+        except OSError:
+            pass
+
+    def close(self) -> None:
+        """Flush and stop the writer thread; safe to call when disabled."""
+        if self._writer_thread is None:
+            return
+        try:
+            self._writer_queue.put(_WRITER_STOP)
+        except Exception:  # noqa: BLE001 - best-effort on teardown
+            pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)